irqtop build fails with `--with-slang`

When trying to build util-linux-2.42 with `--with-slang` configure option, irqtop fails to build:
```
x86_64-pc-linux-gnu-gcc -DHAVE_CONFIG_H -I. -I/var/tmp/portage/sys-apps/util-linux-2.42/work/util-linux-2.42  -include config.h -I/var/tmp/portage/sys-apps/util-linux-2.42/work/util-linux-2.42/include -DLOCALEDIR=\"/usr/share/locale\" -D_PATH_RUNSTATEDIR=\"/run\" -D_PATH_LOCALSTATEDIR=\"/var\" -D_PATH_SYSCONFDIR=\"/etc\" -D_PATH_SYSCONFSTATICDIR=\"/usr/lib\"    -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -fsigned-char -fno-common -Wall -Wextra -Waddress-of-packed-member -Wdiscarded-qualifiers -Wformat-security -Wimplicit-function-declaration -Wmissing-declarations -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-missing-field-initializers -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wsign-compare -Wstrict-prototypes -Wtype-limits -Wuninitialized -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-parameter -Wunused-result -Wunused-variable -Wvla -Walloca -Werror=sequence-point -I./libsmartcols/src  -march=native -mtune=native -O2 -pipe -c -o sys-utils/irqtop-irqtop.o `test -f 'sys-utils/irqtop.c' || echo '/var/tmp/portage/sys-apps/util-linux-2.42/work/util-linux-2.42/'`sys-utils/irqtop.c
/var/tmp/portage/sys-apps/util-linux-2.42/work/util-linux-2.42/sys-utils/irqtop.c: In function ‘irqtop_printf’:
/var/tmp/portage/sys-apps/util-linux-2.42/work/util-linux-2.42/sys-utils/irqtop.c:101:23: error: implicit declaration of function ‘vw_printw’; did you mean ‘vwprintf’? [-Wimplicit-function-declaration]
  101 |                 ret = vw_printw(ctl->win, fmt, args);
      |                       ^~~~~~~~~
      |                       vwprintf
/var/tmp/portage/sys-apps/util-linux-2.42/work/util-linux-2.42/sys-utils/irqtop.c:101:23: warning: nested extern declaration of ‘vw_printw’ [-Wnested-externs]
/var/tmp/portage/sys-apps/util-linux-2.42/work/util-linux-2.42/sys-utils/irqtop.c:104:35: error: ‘OK’ undeclared (first use in this function)
  104 |         if (!ctl->batch && ret == OK)
      |                                   ^~
/var/tmp/portage/sys-apps/util-linux-2.42/work/util-linux-2.42/sys-utils/irqtop.c:104:35: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [Makefile:14047: sys-utils/irqtop-irqtop.o] Error 1
```

I suppose this is a result of the addition of batch mode (commit 2a989ed5be) which introduced `vw_printw` that - AFAIK - is only available in ncurses and not slang. Unfortunately, `sys-utils/Makemodule.am` contains [this code snippet](https://github.com/util-linux/util-linux/blob/04d22901dd3d91d25e985596d588b9cec1ee267d/sys-utils/Makemodule.am#L72) which omits ncurses support even if `--with-ncurses` or `--with-ncursesw` has been submitted to configure.
Adding `--without-slang` to configure obviously "fixes" the issue.

--- a/configure.ac
+++ b/configure.ac
@@ -1223,6 +1223,10 @@ AS_IF([test "x$have_slang" = xyes || test "x$have_ncursesw" = xyes || test "x$ha
 		     AC_DEFINE(HAVE_RESIZETERM, 1,
 			       [Define if curses library has the resizeterm().])
 	])
+	AC_CHECK_LIB([$CURSES_LIB_NAME], vw_printw, [
+		     AC_DEFINE(HAVE_VW_PRINTW, 1,
+			       [Define if curses library has the vw_printw().])
+	])
 ])
 
 

--- a/meson.build
+++ b/meson.build
@@ -320,6 +320,8 @@ foreach curses_libs : [lib_slang, lib_ncursesw, lib_ncurses]
     conf.set('HAVE_USE_DEFAULT_COLORS', have ? 1 : false)
     have = cc.has_function('resizeterm', dependencies : curses_libs)
     conf.set('HAVE_RESIZETERM', have ? 1 : false)
+    have = cc.has_function('vw_printw', dependencies : curses_libs)
+    conf.set('HAVE_VW_PRINTW', have ? 1 : false)
     break
   endif
 endforeach

--- a/sys-utils/irqtop.c
+++ b/sys-utils/irqtop.c
@@ -101,6 +101,22 @@ static inline int vw_printw(WINDOW *win, const char *fmt, va_list args)
 }
 #endif
 
+/* Write pre-formatted data to the screen, followed by @nl newlines */
+static inline void irqtop_puts(struct irqtop_ctl *ctl, const char *data, int nl)
+{
+	if (ctl->batch) {
+		fputs(data, stdout);
+		while (nl--)
+			fputc('\n', stdout);
+	} else {
+		waddstr(ctl->win, (char *) data);
+		while (nl--)
+			waddch(ctl->win, '\n');
+		wrefresh(ctl->win);
+	}
+}
+
+/* Write formatted output to the screen */
 static inline int irqtop_printf(struct irqtop_ctl *ctl, const char *fmt, ...)
 {
 	int ret = 0;
@@ -184,7 +200,7 @@ static int update_screen(struct irqtop_ctl *ctl, struct irq_output *out)
 	if (cpus) {
 		scols_print_table_to_string(cpus, &data);
 		if (data && *data)
-			irqtop_printf(ctl, "%s\n\n", data);
+			irqtop_puts(ctl, data, 2);
 		free(data);
 	}
 
@@ -199,14 +215,14 @@ static int update_screen(struct irqtop_ctl *ctl, struct irq_output *out)
 		*p = '\0';
 		if (!ctl->batch)
 			attron(A_REVERSE);
-		irqtop_printf(ctl, "%s\n", data);
+		irqtop_puts(ctl, data, 1);
 		if (!ctl->batch)
 			attroff(A_REVERSE);
 		data = p + 1;
 	}
 
 	if (data && *data)
-		irqtop_printf(ctl, "%s\n\n", data);
+		irqtop_puts(ctl, data, 2);
 	free(data0);
 
 	/* clean up */

--- a/sys-utils/irqtop.c
+++ b/sys-utils/irqtop.c
@@ -86,6 +86,21 @@ struct irqtop_ctl {
 		softirq;
 };
 
+#ifndef OK
+# define OK 0
+#endif
+
+#ifndef HAVE_VW_PRINTW
+static inline int vw_printw(WINDOW *win, const char *fmt, va_list args)
+{
+	char buf[BUFSIZ];
+
+	if (vsnprintf(buf, sizeof(buf), fmt, args) < 0)
+		return ERR;
+	return waddstr(win, buf);
+}
+#endif
+
 static inline int irqtop_printf(struct irqtop_ctl *ctl, const char *fmt, ...)
 {
 	int ret = 0;

